stock_picking_auto_create_lot_biochousa/models/stock_picking.py

```diff
@@ -0,0 +1,57 @@
+# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models, _
+from datetime import datetime
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def button_validate(self):
+
+        if self.picking_type_id.auto_create_lot:
+            for line in self.move_line_ids.filtered(
+                lambda x: (
+                    not x.lot_id
+                    and not x.lot_name
+                    and x.product_id.tracking != "none"
+                    and x.product_id.auto_create_lot
+                )
+            ):
+                # Create lot_name as Id.Product+Contact/Warehouse code+Date+1/0(ECO)
+                # Code Product = Stock_move_line.product_id --> Product_product.default_code
+                if line.product_id.default_code:
+                    product_code = line.product_id.default_code
+                else:
+                    raise UserError(_('Cannot find the internal reference of product.'))
```

Comment:
Please add product name to error message, in orden to clarify which product has the problem. If the picking contains several product this should be useful